compiler/rustc_middle/src/arena.rs

```diff
@@ -14,6 +14,7 @@ macro_rules! arena_types {
             [] layouts: rustc_target::abi::Layout,
             // AdtDef are interned and compared by address
             [] adt_def: rustc_middle::ty::AdtDef,
+            [] steal_thir: rustc_data_structures::steal::Steal<rustc_middle::thir::Thir<$tcx>>,
```

Comment:
I wonder if it would be worth making this a `Box<Thir>`, so that we don't store as many pointers inline in the map.